qualtran/bloqs/arithmetic/bitwise.py

```diff
@@ -118,3 +123,73 @@ def _cxork() -> XorK:
     cxork = XorK(QUInt(8), 0b01010111).controlled()
     assert isinstance(cxork, XorK)
     return cxork
+
+
+@frozen
+class Xor(SpecializedSingleQubitControlledGate):
+    """Xor the value of one register into another via CNOTs.
+
+    When both registers are in computational basis and the destination is 0,
+    effectively copies the value of the source into the destination.
+
+    Args:
+        dtype: Data type of the input registers `x` and `y`.
+        control_val: An optional single bit control, apply the operation when
+            the control qubit equals the `control_val`.
+
+    Registers:
+        x: The source register.
+        y: The target register.
+    """
+
+    dtype: QDType
+    control_val: Optional[int] = None
+
+    @cached_property
+    def signature(self) -> Signature:
+        return Signature.build_from_dtypes(x=self.dtype, y=self.dtype)
+
+    @cached_property
+    def control_registers(self) -> Tuple[Register, ...]:
+        if self.control_val is not None:
+            return (Register('ctrl', QBit()),)
+        return ()
+
+    def build_composite_bloq(self, bb: BloqBuilder, x: Soquet, y: Soquet) -> Dict[str, SoquetT]:
+        if not isinstance(self.dtype.num_qubits, int):
+            raise DecomposeTypeError("`dtype.num_qubits` must be a concrete value.")
+
+        xs = bb.split(x)
+        ys = bb.split(y)
+
+        for i in range(len(xs)):
+            xs[i], ys[i] = bb.add_t(CNOT(), ctrl=xs[i], target=ys[i])
+
+        return {'x': bb.join(xs), 'y': bb.join(ys)}
+
+    def build_call_graph(self, ssa: 'SympySymbolAllocator') -> Set['BloqCountT']:
+        return {(CNOT(), self.dtype.num_qubits)}
+
```

Comment:
You need to use a `Toffoli()` instead of a `CNOT` whenever `self.control_val` is not None. So this decomposition would be incorrect.  

I think I take my suggestion back -- for a simple bloq like this, it's probably best to just let the `Controlled()` bloq handle the controlled version and it's cost instead of explicitly defining the controlled bloq. The advantages of having a custom controlled Bloq, as I see it, is mainly 
a) More control over diagrams
b) Explicit control over decomposition and costings, though in this case that shouldn't matter.

@anurudhp Do you remember why you explicitly defined the controlled version for `XorK` in https://github.com/quantumlib/Qualtran/pull/1110/files?

Reply:
Falling back to `Controlled()`.

Reply:
Actually not sure, I think I had it initially to match `AddK`, but forgot to remove it later.

Reply:
I think it's best to avoid specialized controls unless we want to explicitly specify the diagrams for `Xor` and `XorK`. I'll cleanup the `XorK` bloq.